common/otel_metrics.go

```diff
@@ -0,0 +1,138 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"context"
+	"errors"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/metric"
+)
+
+var (
+	fsOpsMeter     = otel.Meter("fs_op")
+	gcsMeter       = otel.Meter("gcs")
+	fileCacheMeter = otel.Meter("file_cache")
+)
+
+type otelMetrics struct {
+	fsOpsCount      metric.Int64Counter
+	fsOpsErrorCount metric.Int64Counter
+	fsOpsLatency    metric.Float64Histogram
+
+	gcsReadCount          metric.Int64Counter
+	gcsReadBytesCount     metric.Int64Counter
+	gcsReaderCount        metric.Int64Counter
+	gcsRequestCount       metric.Int64Counter
+	gcsRequestLatency     metric.Float64Histogram
+	gcsDownloadBytesCount metric.Int64Counter
+
+	fileCacheReadCount      metric.Int64Counter
+	fileCacheReadBytesCount metric.Int64Counter
+	fileCacheReadLatency    metric.Float64Histogram
```

Comment:
@kislaykishore IIUC this is the exhaustive list of all metrics we emit from gcsfuse, If some new metric needs to be added, it ll be added here ? If yes, Can you add corresponding comment

Also, with more metrics, a single struct will be difficult to scale, we should consider the refactoring for breaking this up domain wise

Reply:
I've added a comment.
Since this is not an exported struct, I think we can easily refactor it as soon as we find it becoming unmanageable.